process/peer/process.go

```diff
@@ -661,6 +657,24 @@ func (vs *validatorStatistics) ResetValidatorStatisticsAtNewEpoch(vInfos map[uin
 	return nil
 }
 
+func (vs *validatorStatistics) setToJailedIfNeeded(
+	peerAccount state.PeerAccountHandler,
+	validator *state.ValidatorInfo,
+) {
+	if !vs.flagJailedEnabled.IsSet() {
+		return
+	}
+
+	if validator.List == string(core.JailedList) && peerAccount.GetList() != string(core.JailedList) {
+		peerAccount.SetListAndIndex(validator.ShardId, string(core.JailedList), validator.Index)
+		return
+	}
+
+	if peerAccount.GetUnStakedEpoch() == core.DefaultUnstakedEpoch && peerAccount.GetList() == string(core.InactiveList) {
+		peerAccount.SetListAndIndex(validator.ShardId, string(core.JailedList), validator.Index)
```

Comment:
The same set here and in line 669. Why not extract both conditions in shouldBeJailed variable and reduce the code

Reply:
Not exactly the same. Would leave as it is

Reply:
It is only 1 line of code.